pkg/pathconfig/provider.go

```diff
@@ -0,0 +1,66 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package pathconfig
+
+import (
+	"github.com/ghodss/yaml"
+	"sigs.k8s.io/kustomize/pkg/loader"
+	"sigs.k8s.io/kustomize/pkg/pathconfig/defaultpath"
+)
+
+type pathConfigProvider struct {
+	t TransformerConfig
+}
+
+// NewPathconfigProvider returns pathConfigProvider
```

Comment:
Perhaps just two factory methods
```
func MakeTransformerConfigFromFile(
    l loader, paths []string) *TransformerConfig
```
which calls
```
func MakeTransformerConfigFromBytes(
    x []byte) *TransformerConfig
```

and this latter is the same func used to create the default from the const string (cast the string to bytes in the call).


Then define this to return a new object, leaving the old one alone:
```
func (*TransformerConfig) Merge(
   *TransformerConfig) *TransformerConfig
```
because immutable objects are safer than mutable ones and we're not concerned about performance here.

The tests for these are simple, simpler than testing `NewPathConfigProvider`

Reply:
updated according to this. Also removed provider type since it looks awkward.